agent/agent.go

```diff
@@ -285,6 +285,9 @@ func (a *Agent) run(ctx context.Context) {
 				if err := a.worker.Assign(ctx, msg.Changes); err != nil {
 					log.G(ctx).WithError(err).Error("failed to synchronize worker assignments")
 				}
+				// then, call Listen on the worker again to update all of the
+				// task statuses.
+				a.worker.Report(ctx, reporter)
```

Comment:
So, we trigger this on a full, incoming set? This will block the agent loop. Can we start a goroutine here?